src/main/java/dev/openfeature/sdk/OpenFeatureClient.java

```diff
@@ -48,18 +43,29 @@ public class OpenFeatureClient implements Client {
      * @param domain         An identifier which logically binds clients with providers (used by observability tools).
      * @param version        Version of the client (used by observability tools).
      * @deprecated Do not use this constructor. It's for internal use only.
-     *             Clients created using it will not run event handlers.
-     *             Use the OpenFeatureAPI's getClient factory method instead.
+     *     Clients created using it will not run event handlers.
+     *     Use the OpenFeatureAPI's getClient factory method instead.
      */
     @Deprecated() // TODO: eventually we will make this non-public. See issue #872
-    public OpenFeatureClient(OpenFeatureAPI openFeatureAPI, String domain, String version) {
+    public OpenFeatureClient(
+            ProviderAccessor providerAccessor,
```

Comment:
`ProviderAccessor` is exposed outside its defined visibility scope. `OpenFeatureClient` and the constructor are public, but the `ProviderAccessor` is package privet.

Today I learned... I would have expected the compiler to not allow such constructs.

Reply:
How should I handle this? Should I make the constructor package private, as it is deprecated anyway?

Reply:
I would add a package internal accessor to `getProviderStateManager(String domain)`. Then you can access the state manager in the client and you can remove the ´ProviderAccessor` interface completely.